### ISSUE TITLE
Cant open Reddit.com in an incognito window.

### ISSUE THREAD

**Issue by Angxddeep** [brave/brave-browser#42605]
### Description

This sounds a bit absurd, but I can't open reddit.com in an incognito window it opens up fine in a normal window, the issue is with an incognito window specifically. I tried restarting my OS and even checked DNS and there was no issue there as well, I have attached a screen recording for reference. Also, I am using the flatpak version of Brave, So I didn't know how to take appropriate logs. 
https://github.com/user-attachments/assets/f5e1dd34-c216-4431-b820-d4881405d0e6

### Steps to reproduce

1. Download Brave Flatpak version
2. Open reddit.com in an incognito window 

### Actual result

webpage doesnt load

### Expected result

webpage should load

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

Brave	1.73.91 Chromium: 131.0.6778.85 (Official Build) unknown (64-bit) 
Revision	35427fa72a008a7e861fa9828e07a0ba2cd46ebd
OS	Linux

### Channel information

- [x] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by ryanbr** [comment]
What error message is shown? @Angxddeep 
```
You've been blocked by network security.
To continue, log in to your Reddit account or use your developer token

If you think you've been blocked by mistake, file a ticket below and we'll look into it.
```

This is reddit checking for this, occurs in other browsers

**Comment by Angxddeep** [comment]
> What error message is shown? [@Angxddeep](https://github.com/Angxddeep)
There is no error message, it just keeps on loading in a weird way

**Comment by Angxddeep** [comment]
This got fixed with 1.73.97 release, I don't know why it happened and how it got fixed 🤔
